src/commands/modping.ts

```diff
@@ -0,0 +1,160 @@
+    }
+
+    let message = 'An error occurred while processing your request.';
+
+    try {
+        const subcommandGroup = interaction.options.getSubcommandGroup(false);
+        const subcommand = interaction.options.getSubcommand();
+
+        if (subcommand === 'toggle') {
+            const member = interaction.member as GuildMember;
+            const role = member.guild.roles.cache.get(roleId);
+
+            if (role) {
+                if (member.roles.cache.has(roleId)) {
+                    await member.roles.remove(role);
+                    message = `<@&${roleId}> has been removed from you.`;
+                } else {
+                    await member.roles.add(role);
+                    message = `<@&${roleId}> has been assigned to you.`;
+                }
+
+                const settings = await ModPingSettings.findOne({ where: { user_id: member.id } });
+                if (settings) {
+                    message += '\nAuto-assign will override this if you change statuses.';
+                }
+            } else {
+                message = 'Role not found.';
+            }
+        } else if (subcommandGroup === 'auto') {
+            const userId = interaction.user.id;
+            if (subcommand === 'assign') {
+                const online = interaction.options.getBoolean('online') ?? true;
+                const idle = interaction.options.getBoolean('idle') ?? true;
+                const dnd = interaction.options.getBoolean('do_not_disturb') ?? false;
+                const offline = interaction.options.getBoolean('offline') ?? false;
+                const statusList = [];
+                if (online) {
+                    statusList.push('Online');
+                }
+                if (idle) {
+                    statusList.push('Idle');
+                }
+                if (dnd) {
+                    statusList.push('Do Not Disturb');
+                }
+                if (offline) {
+                    statusList.push('Offline');
+                }
+
+                if (statusList.length === 0 || statusList.length === 4) {
+                    await interaction.reply({ content: `Sorry, this setup won't work. You need to have at least one status that isn't the same as the rest.`, ephemeral: true });
+                    return;
+                } else if (statusList.length === 1) {
+                    message = `<@&${roleId}> will be assigned when you are ${statusList[0]}.`;
+                } else {
+                    const lastStatus = statusList.pop();
+                    message = `<@&${roleId}> will be assigned when you are ${statusList.join(', ')} or ${lastStatus}.`;
+                }
+
+                const settings = {
+                    online,
+                    idle,
+                    dnd,
+                    offline
+                };
+
+                await ModPingSettings.upsert({
+                    user_id: userId,
+                    settings: JSON.stringify(settings)
```

Comment:
It's better to make 4 separate boolean columns for the 4 statuses instead of putting JSON in the database.

Reply:
fair, can do that